x-pack/plugins/cloud_security_posture/server/routes/status/status.ts

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { transformError } from '@kbn/securitysolution-es-utils';
+import type { ElasticsearchClient, SavedObjectsClientContract } from '@kbn/core/server';
+import type {
+  AgentPolicyServiceInterface,
+  AgentService,
+  PackagePolicyServiceInterface,
+  PackageService,
+} from '@kbn/fleet-plugin/server';
+import type { GetAgentPoliciesResponseItem } from '@kbn/fleet-plugin/common';
+import moment, { MomentInput } from 'moment';
+import {
+  CLOUD_SECURITY_POSTURE_PACKAGE_NAME,
+  INFO_ROUTE_PATH,
+  LATEST_FINDINGS_INDEX_DEFAULT_NS,
+} from '../../../common/constants';
+import type { CspAppContext } from '../../plugin';
+import type { CspRouter } from '../../types';
+import type { CspSetupStatus, Status } from '../../../common/types';
+import {
+  addRunningAgentToAgentPolicy,
+  getCspAgentPolicies,
+  getCspPackagePolicies,
+} from '../../lib/fleet_util';
+
+export const INDEX_TIMEOUT_IN_MINUTES = 10;
+
+const isFindingsIndexExists = async (esClient: ElasticsearchClient): Promise<boolean> => {
+  try {
+    const queryResult = await esClient.search({
+      index: LATEST_FINDINGS_INDEX_DEFAULT_NS,
+      query: {
+        match_all: {},
+      },
+      size: 1,
+    });
```

Comment:
reverted back from using `esclient.indices.exists`, the check is a bit different and it's up to `transforms` workflow which I prefer not to mess with